src/main/java/org/truffleruby/core/rope/RopeConstants.java

```diff
@@ -126,4 +139,38 @@ public static LeafRope lookupUSASCII(String string) {
         }
     }
 
+    private static final LeafRope[] PADDED_NUMBERS = new LeafRope[100];
+
+    static {
+        for (int n = 0; n < 10; n++) {
+            PADDED_NUMBERS[n] = UTF8_SINGLE_BYTE_ROPES['0' + n];
+        }
+
+        for (int n = 10; n < 100; n++) {
+            PADDED_NUMBERS[n] = new AsciiOnlyLeafRope(new byte[]{ '0', (byte) ('0' + n) }, UTF8Encoding.INSTANCE);
```

Comment:
This seems incorrect, the first digit is not always 0, right?

Reply:
Oh duh sorry I've done this bit completely wrong... will fix